ngraph/core/src/op/range.cpp

```diff
@@ -26,6 +26,215 @@
+}
+
+bool ngraph::op::v4::Range::visit_attributes(AttributeVisitor& visitor)
+{
+    visitor.on_attribute("output_type", m_output_type);
+    return true;
+}
+
+void op::v4::Range::validate_and_infer_types()
+{
+    set_input_is_relevant_to_shape(0);
+    set_input_is_relevant_to_shape(1);
+    set_input_is_relevant_to_shape(2);
+
+    NODE_VALIDATION_CHECK(
+        this, get_input_partial_shape(0).compatible(Shape{}), "'start' input is not a scalar");
+    NODE_VALIDATION_CHECK(
+        this, get_input_partial_shape(1).compatible(Shape{}), "'stop' input is not a scalar");
+    NODE_VALIDATION_CHECK(
+        this, get_input_partial_shape(2).compatible(Shape{}), "'step' input is not a scalar");
+
+    auto const_start = as_type_ptr<op::Constant>(this->input_value(0).get_node_shared_ptr());
+    auto const_stop = as_type_ptr<op::Constant>(this->input_value(1).get_node_shared_ptr());
+    auto const_step = as_type_ptr<op::Constant>(this->input_value(2).get_node_shared_ptr());
+
+    double start = 0;
+    double stop = 0;
+    double step = 0;
+
+    if (const_start != nullptr)
+    {
+        std::vector<double> start_val = const_start->cast_vector<double>();
+        NODE_VALIDATION_CHECK(this, start_val.size() == 1);
+        start = start_val[0];
+        NODE_VALIDATION_CHECK(
+            this, std::isfinite(start) && !std::isnan(start), "'start' cannot be nan or infinite.");
+    }
+
+    if (const_stop != nullptr)
+    {
+        std::vector<double> stop_val = const_stop->cast_vector<double>();
+        NODE_VALIDATION_CHECK(this, stop_val.size() == 1);
+        stop = stop_val[0];
+        NODE_VALIDATION_CHECK(
+            this, std::isfinite(stop) && !std::isnan(stop), "'stop' cannot be nan or infinite.");
+    }
+
+    if (const_step != nullptr)
+    {
+        std::vector<double> step_val = const_step->cast_vector<double>();
+        NODE_VALIDATION_CHECK(this, step_val.size() == 1);
+        step = step_val[0];
+        NODE_VALIDATION_CHECK(
+            this, std::isfinite(step) && !std::isnan(step), "'step' cannot be nan or infinite.");
+    }
+
+    PartialShape result{PartialShape::dynamic(1)};
+
+    if (const_start != nullptr && const_stop != nullptr && const_step != nullptr)
+    {
+        double span;
+
+        if ((step > 0 && start >= stop) || (step < 0 && start <= stop))
+        {
+            span = 0;
+        }
+        else
+        {
+            span = stop - start;
+        }
+
+        double strided = ceil(fabs(span) / fabs(step));
+
+        result = PartialShape{Dimension(static_cast<int64_t>(strided))};
+    }
+    set_output_type(0, m_output_type, result);
+}
+
+shared_ptr<Node> op::v4::Range::clone_with_new_inputs(const OutputVector& new_args) const
+{
+    check_new_args_count(this, new_args);
+    return make_shared<v4::Range>(new_args.at(0), new_args.at(1), new_args.at(2), m_output_type);
+}
+
+template <typename T>
+bool get_casted_value(const HostTensorPtr& tensor, T* val)
```

Comment:
Why do you need this method?

Looks like you can just call `tensor->get_data_ptr` in [this switch case](https://github.com/openvinotoolkit/openvino/pull/1552/files#diff-5dbe09c03c8c30d7e64fe3928418a106R195).

It should help avoid this method and [this if](https://github.com/openvinotoolkit/openvino/pull/1552/files#diff-5dbe09c03c8c30d7e64fe3928418a106R170).

Reply:
This method is needed since Range should work even if start, stop and step tensors have different types.

Reply:
But at the current moment all tensors don't have different types